src/main/java/ro/developmentfactory/thedrones/service/DronePositionServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package ro.developmentfactory.thedrones.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.developmentfactory.thedrones.controller.dto.DroneStatusResponse;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+import ro.developmentfactory.thedrones.repository.DroneStatusRepository;
+import ro.developmentfactory.thedrones.repository.entity.Drone;
+import ro.developmentfactory.thedrones.repository.entity.DroneStatus;
+
+import java.util.UUID;
+
+@Service
+public class DronePositionServiceImpl implements DronePositionService {
+
+    private final DroneStatusRepository droneStatusRepository;
+    private final DroneRepository droneRepository;
+    private static final Logger log = LoggerFactory.getLogger(DronePositionServiceImpl.class);
+
+    public DronePositionServiceImpl(DroneStatusRepository droneStatusRepository, DroneRepository droneRepository) {
+        this.droneStatusRepository = droneStatusRepository;
+        this.droneRepository = droneRepository;
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turnDroneLeft(DroneStatusResponse droneStatusResponse) {
+        log.debug("Turning drone left :{}",droneStatusResponse);
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus currentStatus = drone.getDroneStatusList().getFirst();
+        currentStatus.setFacingDirection(currentStatus.getFacingDirection().turnLeft());
+
+        droneStatusRepository.save(currentStatus);
+       return convertToResponse(currentStatus);
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turnDroneRight(DroneStatusResponse droneStatusResponse) {
+        log.debug("Turning drone right :{}",droneStatusResponse);
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus currentStatus = drone.getDroneStatusList().getFirst();
+        currentStatus.setFacingDirection(currentStatus.getFacingDirection().turnRight());
+
+        droneStatusRepository.save(currentStatus);
+        return convertToResponse(currentStatus);
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse moveForward(DroneStatusResponse droneStatusResponse) {
```

Comment:
this method looks too complex and hard to read. it has 2 parts: validation and position update. I'd recommend you extracting validation to a separate method and calling it first - you don't even need to bother calculating any new positions if the move is not valid. there are multiple ways to validate in spring boot, the easiest is to have a set of cases when the move is invalid in dependence of X, Y, and direction. if validation passes, you can update the position - you should be sure that the move is valid anyway. you can extract 2nd part to a separate method too